Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -1,5 +1,71 @@
 ## iOS Platform APIs
 
+### Testing App Permissions
+
+#### Overview
+iOS makes all mobile applications run under the `mobile` user. Each application is sandboxed and limited using policies enforced by the Trusted BSD mandatory access control framework. These policies are called profiles and all third-party applications use on generic sandbox profile: the container permission list. <TODO: ADD REFERENCE TO
+http://www.icri-sc.org/fileadmin/user_upload/Group_TRUST/PubsPDF/sandscout-final-ccs-2016.pdf >.
+
+On iOS, apps need to request permission to the user for accessing one of the following data or resources: Bluetooth peripherals, Calendar data, Camera, Contacts, Health sharing, Health updating, HomeKit, Location, Microphone, Motion, Music and the media library, Photos, Reminders, Siri, Speech recognition, and the TV provider.<TODO: add reference to https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/ExpectedAppBehaviors/ExpectedAppBehaviors.html#//apple_ref/doc/uid/TP40007072-CH3-SW7 >
+Even though Apple urges through various sources (such as <https://developer.apple.com/library/archive/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/ExpectedAppBehaviors/ExpectedAppBehaviors.html#//apple_ref/doc/uid/TP40007072-CH3-SW7>) to protect the privacy of the user, it can still be the case that an app requests too many permissions.
+
+Next to the resources for which permission is requested, there is a set of capabilities, which can be required by the app developer in order to run the device. These capabilities (`UIRequiredDeviceCapabilities`) are listed at <TODO ADD REFERENCE TO https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/iPhoneOSKeys.html#//apple_ref/doc/uid/TP40009252-SW1 here!>. These capabilities are used by App Store and by iTunes to ensure that only compatible devices are listed. They often do not provide implicit permission (TODO: EXPLAIN WHAT IMPLICIT IS: NO USER INTERACTION THERE!). <TODO: CHECK FOR WHICH THIS IS DIFFERENT, SUCH AS INTER-AP COMMUNICATION>. Note that the actual available capabilities differ per type of developer profile used to sign the application (https://developer.apple.com/support/app-capabilities/).
+
+#### Static analysis
+
+Since iOS 10, there are three areas which you need to inspect for permssions:
+- the info.plist file,
+- the `<appname>.enttitlements` file, where <appname> is the name of the application, or the
```

Comment:
remove  "the" at the end of the the line

Reply:
thanks!